src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2947,22 +2947,35 @@ def aks_mesh_get_revisions(
     return None
 
 
+def check_iterator(iterator):
+    import itertools
+    try:
+        first = next(iterator)
+    except StopIteration:   # iterator is empty
+        return True, iterator
+    except TypeError:       # iterator is not iterable, e.g. None
+        return True, iterator
+    return False, itertools.chain([first], iterator)
+
+
 def aks_mesh_get_upgrades(
         cmd,
         client,
         resource_group_name,
         name
 ):
     upgradeProfiles = client.list_mesh_upgrade_profiles(resource_group_name, name)
-    # 'upgradeProfiles' is an ItemPaged object
+    is_empty, upgradeProfiles = check_iterator(upgradeProfiles)
+    if is_empty:
+        logger.warning("No mesh upgrade profiles found for the cluster '%s' " +
+                       "in the resource group '%s'.", name, resource_group_name)
+        return None
```

Comment:
When will the behavior of returning null be changed on the server side?

Reply:
we are actively discussing the fix within the team. 
ADO Task - [Bug 27079429](https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/27079429): Incorrect response code for `az aks mesh get-updates` command for a cluster with ASM disabled